### PR TITLE
Adds the adminspawn debug outfit

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

This a port from TG and a few other codebases. The debug outfit is a "small" admin-spawn only outfit, providing all the tools you could ever want to debug stuff. (well, most of it.)

This includes
- A headset with all comms (yes. all comms, even medical 1 and security 1)
- vomit prevention welding mask (wow)
- a resurrection and death wand, both recharging and with 500 charges
- advanced glasses, with all HUDs built in, and allowing you to toggle X-RAY and disabling all darkness
- A new human spawner object, for spawning "test subjects"
- An advanced RCD, with infinite matter
- 50000 credits
- An advanced surgical implant
- An advanced janitorial implant (it also has a tazer, for shits and giggles.)
- World's fastest cleaning tools, and farthest reaching.
<details>
  <summary>Thats the new stuff out of the way, heres the rest of the debug stuff if you care</summary>
  
 - emag
 - geiger counter
 - bluespace RPD
 - 7 bluespace beakers
 - ALL THE MATERIALS.
 - a centcomm officer's box, for the pinpointers, internals, combat meds, etc
 - an admin uplink (2500 tc)
 - a badmin book (summon a cookie, or whatever you var edit.)
 - badmin vodka, because yes.
 - jaws of life (i use these during testing sometimes)
 - some other silly admin only stuff, so its easily accessible if you want to mess around.
  
</details>

The debug module also has a few more of the modules, advanced stealth, advanced magboots, and advanced jetpack, along with having the honk module pinned by default (honk). The speed on the suit is also 0.1 seconds per section now, now thats a fast sealing modsuit (useful for when you need to take the modsuit off to do something else)

This also fixes some issues
- Deathsquad can now properly spawn with their guns on their back again, because I changed the equip system slightly
- If credits are somehow over their max limit, they will display the proper amount, not 10000

## Why It's Good For The Game
Makes testing more easier to set up, and provides some fun tools for new developers to use

## Images of changes

image
image
image
image
image
image
image

## Testing

Spawned in as debug outfit, deathsquad, and other roles with modsuits starting on them as default, worked as intended. Messed around with the money, it stayed the proper amount (I still dont understand why that check was there).
Everything new in this PR worked as intended, except the syndikey stuff in the headset but... I couldn't figure that out. I'd appreciate some help on that.

## Changelog
Not intended for players to use.

### CODE REVIEW

File: code/datums/outfits/outfit_debug.dm [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+	uniform = /obj/item/clothing/under/costume/patriotsuit
+	back = /obj/item/mod/control/pre_equipped/debug
+	backpack_contents = list(
+		/obj/item/melee/energy/axe = 1,
+		/obj/item/storage/part_replacer/bluespace/tier4 = 1,
+		/obj/item/gun/magic/wand/resurrection/debug = 1,
+		/obj/item/gun/magic/wand/death/debug = 1,
+		/obj/item/debug/human_spawner = 1,
+)
```
Comment (Burzah) [inline code/datums/outfits/outfit_debug.dm:12]:
```suggestion
	)
```

File: code/datums/outfits/outfit_debug.dm [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	name = "Debug outfit"
+
+	uniform = /obj/item/clothing/under/costume/patriotsuit
+	back = /obj/item/mod/control/pre_equipped/debug
+	backpack_contents = list(
+		/obj/item/melee/energy/axe = 1,
+		/obj/item/storage/part_replacer/bluespace/tier4 = 1,
+		/obj/item/gun/magic/wand/resurrection/debug = 1,
+		/obj/item/gun/magic/wand/death/debug = 1,
+		/obj/item/debug/human_spawner = 1,
```
Comment (Burzah) [inline code/datums/outfits/outfit_debug.dm:11]:
```suggestion
		/obj/item/debug/human_spawner = 1
```

**Review by Burzah — COMMENTED** [review]
Just some small formatting stuff to match your other lists.

File: code/datums/outfits/outfit_debug.dm [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+	shoes = /obj/item/clothing/shoes/combat/swat
+
+	// shit down here is mine
+	box = /obj/item/storage/box/debug/debugtools
+	suit_store = /obj/item/tank/internals/oxygen
+	gloves = /obj/item/clothing/gloves/combat
+	id = /obj/item/card/id/admin
+	pda = /obj/item/pda/centcom
+
+	internals_slot = slot_s_store //? make sure this is right
```
Comment (lewcc) [inline code/datums/outfits/outfit_debug.dm:26]: reads like a TODO, is it?

File: code/datums/outfits/outfit_debug.dm [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+/obj/item/reagent_containers/spray/cleaner/advanced/debug
+	name = "AVD-CNED advanced space cleaner"
+	desc = "AVD-CNED!-brand non-foaming space cleaner! How fancy."
+	volume = 50000
+	spray_maxrange = 10
+	spray_currentrange = 10
+	list_reagents = list("cleaner" = 50000)
+	delay = 0.1 SECONDS // it costs 1000 reagents to fire this cleaner... for 12 seconds.
+
+
```
Comment (lewcc) [inline code/datums/outfits/outfit_debug.dm:225]: Honestly you can probably move the debug IMS in here as well for consistency

File: code/datums/outfits/outfit_debug.dm [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/human_user = user
+	if(human_user.glasses != src)
+		return
+	if(xray)
+		remove_xray(human_user)
+	else
+		add_xray(human_user)
+	xray = !xray
```
Comment (lewcc) [inline code/datums/outfits/outfit_debug.dm:121]: Add a to_chat here so it's clear it did something

**Review by lewcc — APPROVED** [review]
Looks fine

**Pull request merged**